### ISSUE TITLE
az login broken with service principal

### ISSUE THREAD

**Issue by wbuchwalter** [Azure/azure-cli#2548]
Trying to login with a service principal is now failing:

`az login --service-principal -u 'xxxx' -p 'xxxx' -t 'xxxx'`
```
File "/azure-cli/src/azure-cli-core/azure/cli/core/_profile.py", line 485, in save_service_principal_cred
    self._service_principal_creds.pop(matched[0])
TypeError: 'dict' object cannot be interpreted as an integer
```

This scenario works well when checking out at the commit directly before https://github.com/Azure/azure-cli/pull/2457 @yugangw-msft.

Thanks!

**Comment by yugangw-msft** [comment]
I am not able to reproduce it. 
If you run `az account clear` and redo it, does that repro?

**Comment by wbuchwalter** [comment]
If i run `az account clear` the error goes away.

Thanks!

**Comment by mikebridge** [comment]
I suspect this is the same situation I get, but with a slightly different error: `TypeError: an integer is required`. 

```
an integer is required
Traceback (most recent call last):
  File "/home/bridge/lib/azure-cli/local/lib/python2.7/site-packages/azure/cli/main.py", line 37, in main
    cmd_result = APPLICATION.execute(args)
  File "/home/bridge/lib/azure-cli/local/lib/python2.7/site-packages/azure/cli/core/application.py", line 157, in execute
    result = expanded_arg.func(params)
  File "/home/bridge/lib/azure-cli/local/lib/python2.7/site-packages/azure/cli/core/commands/__init__.py", line 368, in _execute_command
    raise ex
TypeError: an integer is required
```

In case someone's googling the error message, running `az account clear` will also fix this problem.

**Comment by yugangw-msft** [comment]
The original issue will be fixed through #2848.  `an integer is required` appears a different error case, please open a new issue with repro, and I will take a look
